src/greentea/gtea/greentea_hooks.py

```diff
@@ -35,36 +36,27 @@ def __init__(self, name, cmd):
         GreenteaTestHook.__init__(self, name)
         self.cmd = cmd
 
-    def run_cli_process(self, cmd):
-        """! Runs command as a process and return stdout, stderr and ret code
-        @param cmd Command to execute
-        @return Tuple of (stdout, stderr, returncode)
-        """
-        _stdout, _stderr, ret = None, None, -1
-        try:
-            p = Popen(cmd, stdout=PIPE, stderr=PIPE, shell=True)
-            _stdout, _stderr = p.communicate()
-            ret = p.returncode
-        except OSError as e:
-            gt_logger.gt_log_err(str(e))
-            ret = -1
-        return _stdout, _stderr, ret
-
     def run(self, format=None):
         """! Runs hook after command is formated with in-place {tags}
         @format Pass format dictionary to replace hook {tags} with real values
         @param format Used to format string with cmd, notation used is e.g: {build_name}
         """
         gt_logger.gt_log("hook '%s' execution" % self.name)
         cmd = self.format_before_run(self.cmd, format)
-        gt_logger.gt_log_tab("hook command: %s" % cmd)
-        (_stdout, _stderr, ret) = self.run_cli_process(cmd)
-        if _stdout:
-            print(_stdout)
-        if ret:
-            gt_logger.gt_log_err("hook exited with error: %d, dumping stderr..." % ret)
-            print(_stderr)
-        return ret
+        gt_logger.gt_log_tab(f"hook command: {cmd}")
+        try:
+            p = run(self.cmd, capture_output=True, shell=True)
+        except OSError as e:
+            gt_logger.gt_log_err(str(e))
+            return -1
+        if p.stdout:
```

Comment:
Why not just set `capture_output=False`? We seem to print `stdout` and `stderr` all the time if they aren't empty.

I would also leave `shell=False` unless absolutely necessary, as `shell=True` causes `subprocess` to spawn an intermediate "shell instance" to run the command, which could make us vulnerable to shell injection exploits.

We might have to change `self.cmd` to a list to make this work with `shell=False`, if it's currently just a string. We should probably deal with the shell spawning in another PR anyway as it was preexisting.

Reply:
I've removed the `capture_output` arg. `cmd` is a string, so that will need to be changed to remove `shell=True`.